src/Core/ServerSettings.h

```diff
@@ -176,6 +176,7 @@ namespace DB
     M(Bool, disable_insertion_and_mutation, false, "Disable all insert/alter/delete queries. This setting will be enabled if someone needs read-only nodes to prevent insertion and mutation affect reading performance.", 0) \
     M(UInt64, keeper_multiread_batch_size, 10'000, "Maximum size of batch for MultiRead request to [Zoo]Keeper that support batching. If set to 0, batching is disabled. Available only in ClickHouse Cloud.", 0) \
     M(Bool, use_legacy_mongodb_integration, true, "Use the legacy MongoDB integration implementation. Note: it's highly recommended to set this option to false, since legacy implementation will be removed in the future. Please submit any issues you encounter with the new implementation.", 0) \
+    M(Bool, allow_impersonate_user, true, "Enable/disable the IMPERSONATE feature (EXECUTE AS <user>).", 0) \
```

Comment:
We usually put our experimental feature flags in Core/Settings.cpp, it seems it's better to put this flag there too.